infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -86,4 +87,22 @@
      * Clears all data and resets all segment handlers.
      */
     void reset();
+
+    default OverwriteCause verifyOverwriteCauseForAddress(long address, LogData entry) {
+        LogData oldEntry = read(address);
+        OverwriteCause cause = OverwriteCause.DIFF_DATA;
+
+        if (oldEntry.isHole()) {
+            cause = OverwriteCause.HOLE;
+        } else if (entry.getData() != null && oldEntry.getData() != null) {
+            if (oldEntry.getData().length == entry.getData().length) {
```

Comment:
What if you are writing a hole in an address that has already been hold filled?

Reply:
The cause of overwrite exception would be HOLE, regardless of what you are trying to write, which won't go through due to this exception.